ios/Sources/MeasureSDK/Swift/MeasureInternal.swift

```diff
@@ -175,78 +175,104 @@ final class MeasureInternal { // swiftlint:disable:this type_body_length
     }
 
     func start() {
-        if !isStarted {
-            self.logger.log(level: .info, message: "Starting Measure SDK", error: nil, data: nil)
-            registedCollectors()
-            isStarted = true
-        }
+        guard !isStarted else { return }
+
+        self.logger.log(level: .info, message: "Starting Measure SDK", error: nil, data: nil)
+        registedCollectors()
+        isStarted = true
     }
 
     func stop() {
-        if isStarted && !configProvider.autoStart {
-            self.logger.log(level: .info, message: "Stopping Measure SDK", error: nil, data: nil)
-            unregisterCollectors()
-            isStarted = false
-        }
+        guard isStarted && !configProvider.autoStart else { return }
+
+        self.logger.log(level: .info, message: "Stopping Measure SDK", error: nil, data: nil)
+        unregisterCollectors()
+        isStarted = false
     }
 
     func trackEvent(name: String, attributes: [String: AttributeValue], timestamp: Int64?) {
+        guard isStarted else { return }
+
         customEventCollector.trackEvent(name: name, attributes: attributes, timestamp: timestamp)
     }
 
     func trackEvent(_ name: String, attributes: [String: Any], timestamp: NSNumber?) {
+        guard isStarted else { return }
+
         let transformedAttributes = transformAttributes(attributes)
 
         customEventCollector.trackEvent(name: name, attributes: transformedAttributes, timestamp: timestamp?.int64Value)
     }
 
     func setUserId(_ userId: String) {
+        guard isStarted else { return }
+
         userAttributeProcessor.setUserId(userId)
     }
 
     func clearUserId() {
+        guard isStarted else { return }
+
         userAttributeProcessor.clearUserId()
     }
```

Comment:
These two methods on Android are currently allowed, even if the SDK is not started to allow setting up state when needed. We could allow this even if SDK is not started.